Content.Server/Connection/ConnectionManager.IPIntel.cs

```diff
@@ -0,0 +1,377 @@
+            if (overallTime != null && overallTime.TimeSpent >= _exemptPlaytime)
+            {
+                return (false, string.Empty);
+            }
+        }
+
+        var ip = e.IP.Address;
+
+        // Is this a local ip address?
+        if (IsAddressReservedIpv4(ip) || IsAddressReservedIpv6(ip))
+        {
+            _sawmill.Warning($"{e.UserName} joined using a local address. Do you need IPIntel? Or is something terribly misconfigured on your server?" +
+                             $" Trusting this connection.");
+            return (false, string.Empty);
+        }
+
+        // Check our cache
+        var query = await _db.GetIPIntelCache(ip);
+
+        // Does it exist?
+        if (query != null)
+        {
+            // Skip to score check if result is older than _cacheDays
+            if (DateTime.UtcNow - query.Time <= _cacheDays)
+            {
+                _score = query.Score;
+                return await ScoreCheck(_score, e);
+            }
+        }
+
+        // Check our api limits. If we are ratelimited we back out.
+        HandleRatelimit();
+
+        if (_ratelimitedMinute || _ratelimitedDay || CheckSuddenRateLimit())
+            return _rejectLimited ? (true, Loc.GetString("ipintel-server-ratelimited")) : (false, string.Empty);
+
+        // Ensure our contact email is good to use.
+        if (string.IsNullOrEmpty(_contactEmail) || !_contactEmail.Contains('@') || !_contactEmail.Contains('.'))
+        {
+            _sawmill.Error("IPIntel is enabled, but contact email is empty or not a valid email, treating this connection like an unknown IPIntel response.");
+            return _rejectUnknown ? (true, Loc.GetString("generic-misconfigured")) : (false, string.Empty);
+        }
+
+        _currentRequestsDay++;
+        _currentRequestsMinute++;
+        // Info about flag B: https://getipintel.net/free-proxy-vpn-tor-detection-api/#flagsb
+        // TLDR: We don't care about knowing if a connection is compromised.
+        // We just want to know if it's a vpn. This also speeds up the request by quite a bit. (A full scan can take 200ms to 5 seconds. This will take at most 120ms)
+        using var request = await _http.Client.GetAsync($"{_baseUrl}/check.php?ip={ip}&contact={_contactEmail}&flags={_flags}");
+
+        if (request.StatusCode == HttpStatusCode.TooManyRequests)
+        {
+            _sawmill.Warning("We hit the IPIntel request limit at some point.");
+            CalculateSuddenRatelimit();
+            return _rejectLimited ? (true, Loc.GetString("ipintel-server-ratelimited")) : (false, string.Empty);
+        }
+
+        var response = await request.Content.ReadAsStringAsync();
+        _score = Parse.Float(response);
+
+        if (request.StatusCode == HttpStatusCode.OK)
+        {
+            if (query == null || DateTime.UtcNow - query.Time >= _cacheDays)
```

Comment:
You should always re-save if we have a more up-to-date result, there's no reason to compare against the query time here.